device/common/include/traccc/finding/device/impl/apply_interaction.ipp

```diff
@@ -17,7 +17,7 @@
 namespace traccc::device {
 
 template <typename detector_t>
-TRACCC_DEVICE inline void apply_interaction(
```

Comment:
This (and the other `N` files it is duplicated in), is obviously not really okay.

Perhaps needs a change _somewhere_ to bring this into line with what Alpaka uses, when using an Alpaka back end: https://github.com/alpaka-group/alpaka/blob/1793a0760e57bd3cca2b71170507ae1c26b5c0dd/include/alpaka/core/Common.hpp#L20-L42

This is broadly needed as Alpaka defaults to `__device__ __host__` when you use `ALPAKA_FN_ACC` as it expects...but then you hit warnings / errors of device code being called from the host.

Reply:
I'm not worried about this all too much. We can declare these functions as `TRACCC_HOST_DEVICE`, that should not have negative effects on the native CUDA and HIP builds as far as I'm aware. 🤔